### PR TITLE
Typescript angular2 client

### LINKED ISSUE #1809: angular2 client
Just wondering if there is an angular2 client in the works?

If not, I'll take a crack one.

### PR DESCRIPTION
**Pull request by Vrolijkx**
A working angular2 typescript client based up on @roni-frantchi https://github.com/sapienstech/typescript-angular2-swagger-codegen

With some small changes and added tests.

Will resolve #1809 also makes #2291 obsolete.

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        formParams['{{baseName}}'] = {{paramName}};
+
+{{/formParams}}
+        let requestOptions: RequestOptionsArgs = {
+            method: '{{httpMethod}}',
+            headers: headerParams,
+            search: queryParameters
+        };
+        {{#bodyParam}}
+        requestOptions.body = JSOG.stringify({{paramName}});
```
Comment (roni-frantchi) [inline modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache]: @Vrolijkx seems like you have dropped the `import JSOG` statement, did you forget to swap it here?

**Comment by wing328** [comment]
@Vrolijkx Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR wont' count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

Let me know if you need help fixing it.

File: modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache (hunk 1 above)
Comment (Vrolijkx) [inline modules/swagger-codegen/src/main/resources/typescript-angular2/api.mustache]: You are right will fix it in a moment.

**Comment by Vrolijkx** [comment]
@wing328 Help for fixing the linking is always welcome.

**Comment by wing328** [comment]
@Vrolijkx can you try the following?

https://help.github.com/articles/changing-author-info/

**Comment by Vrolijkx** [comment]
@wing328 Solved it by adding my work email to my account settings.

**Comment by wing328** [comment]
@Vrolijkx now looks good to me.

Can you also create `./bin/typescript-angular2-petstore.sh` similar to `./bin/typescript-angular-petstore.sh` and output the sample to `samples/client/petstore/typescript-angular2`? (remember to git add the new files)

**Comment by Vrolijkx** [comment]
@wing328 Added the samples. But noticed that when running the ```all-petstore.sh``` script that a lot of samples are not up to date.

**Comment by wing328** [comment]
Yup, it's due to the recent restore of the original petstore spec.

I would suggest you to create a new branch for your change moving forward.

**Pull request merged**